### ISSUE TITLE
helmet_detect.py

### ISSUE THREAD

**Issue by Raspberry-moon** [Raspberry-moon/helmet_detect.py#1]
import cv2
import torch
import numpy as np
import pygame
import time
from pathlib import Path
import pathlib
temp = pathlib.PosixPath
pathlib.WindowsPath = pathlib.PosixPath

# YOLOv5 모델 로드 (로컬 경로 사용)
model = torch.hub.load('.', 'custom', path='./best.pt', source='local')

# Pygame 초기화 및 스피커 설정
pygame.mixer.init()
pygame.mixer.music.load("./helmet.mp3")

# 웹캠 설정
cap = cv2.VideoCapture(0)
cap.set(cv2.CAP_PROP_FRAME_WIDTH, 640)  # 해상도 설정
cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 480)

if not cap.isOpened():
    print("웹캠을 열 수 없습니다.")
    exit()

frame_skip = 2  # 프레임 스킵 수
frame_count = 0

while True:
    ret, frame = cap.read()
    if not ret:
        print("프레임을 읽을 수 없습니다.")
        break

    frame_count += 1
    if frame_count % frame_skip != 0:
        continue

    # YOLOv5로 이미지 처리
    results = model(frame)

    # 결과에서 라벨 추출
    labels = results.pandas().xyxy[0]['name'].to_numpy()
    print(f"Detected labels: {labels}")

    # 헬멧 미착용 감지
    if 'helmet' not in labels:
        print("Helmet not detected!")
        if not pygame.mixer.music.get_busy():
            print("Playing warning sound...")
            pygame.mixer.music.play()
    else:
        if pygame.mixer.music.get_busy():
            print("Stopping warning sound...")
            pygame.mixer.music.stop()

    # 결과 이미지 렌더링
    annotated_frame = np.squeeze(results.render())
    cv2.imshow('Helmet Detection', annotated_frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()
